### PR TITLE
Add team to division endpoint

### LINKED ISSUE #26: Endpoint to add a team to a division

### PR DESCRIPTION
**Pull request by adamszcz**
# What I added (link any issues addressed)
Closes #26 

- I also got on a roll and added the ```GetById``` endpoint

# How to test it

Run ``` PUT https://localhost:5001/api/division/<division ID>/add```

Body (team ID):

```
{
    "Id": int
}
```
- See that a full, updated Division object is returned
- Optionally, also run ```GET https://localhost:5001/api/division/<division ID>``` and see that you get the same result as the previous call

### CODE REVIEW

File: CribblyBackend.UnitTests/ControllerTests/DivisionControllerTests.cs
Comment (cszczepaniak) [inline CribblyBackend.UnitTests/ControllerTests/DivisionControllerTests.cs]: 🌵 📖 in .NET5 (once my PR is merged), you can do:
```
private Division _division = new() {...};
```

File: src/CribblyBackend.DataAccess/Exceptions/DivisionNotFoundException.cs
Comment (cszczepaniak) [inline src/CribblyBackend.DataAccess/Exceptions/DivisionNotFoundException.cs]: I think we want one constructor here that takes an ID and constructs a message with it

File: src/CribblyBackend.DataAccess/Repositories/DivisionRepository.cs
Comment (cszczepaniak) [inline src/CribblyBackend.DataAccess/Repositories/DivisionRepository.cs]: 🌵 Can we put the comma on L46?

File: src/CribblyBackend.DataAccess/Repositories/DivisionRepository.cs
Comment (cszczepaniak) [inline src/CribblyBackend.DataAccess/Repositories/DivisionRepository.cs]: 🌵 📖 we can drop the `this.`

File: src/CribblyBackend.DataAccess/Repositories/DivisionRepository.cs
Comment (cszczepaniak) [inline src/CribblyBackend.DataAccess/Repositories/DivisionRepository.cs]: the `else` and its block are unnecessary here; you can just `throw` after the `if`

Comment (cszczepaniak) [reply src/CribblyBackend.DataAccess/Repositories/DivisionRepository.cs]: In fact, I think I'd prefer throwing first if the team doesn't exist, and then executing the UPDATE:
```
if (GetById(id) == null) 
{
    throw new DivisionNotFoundException();
}
// ... do the other stuff
```
This keeps our "happy path" more in the line of sight of the reader of the code

File: src/CribblyBackend/Controllers/DivisionController.cs
Comment (cszczepaniak) [inline src/CribblyBackend/Controllers/DivisionController.cs]: I wonder if it would be better to put this at `{id}/team` and make it a PATCH? I think PUT implies that the _entire_ object is being replaced, while PATCH implies an update of part of an object.... but I also don't really know so 🤷‍♂️

File: CribblyBackend.UnitTests/ControllerTests/DivisionControllerTests.cs
Comment (adamszcz) [reply CribblyBackend.UnitTests/ControllerTests/DivisionControllerTests.cs]: What exactly do you mean by this?

File: src/CribblyBackend.DataAccess/Exceptions/DivisionNotFoundException.cs
Comment (adamszcz) [reply src/CribblyBackend.DataAccess/Exceptions/DivisionNotFoundException.cs]: fixed

File: src/CribblyBackend.DataAccess/Repositories/DivisionRepository.cs
Comment (adamszcz) [reply src/CribblyBackend.DataAccess/Repositories/DivisionRepository.cs]: fixed

File: src/CribblyBackend.DataAccess/Repositories/DivisionRepository.cs
Comment (adamszcz) [reply src/CribblyBackend.DataAccess/Repositories/DivisionRepository.cs]: fixed

File: src/CribblyBackend.DataAccess/Repositories/DivisionRepository.cs
Comment (adamszcz) [reply src/CribblyBackend.DataAccess/Repositories/DivisionRepository.cs]: fixed

File: src/CribblyBackend/Controllers/DivisionController.cs
Comment (adamszcz) [reply src/CribblyBackend/Controllers/DivisionController.cs]: fixed

File: CribblyBackend.UnitTests/ControllerTests/DivisionControllerTests.cs
Comment (cszczepaniak) [reply CribblyBackend.UnitTests/ControllerTests/DivisionControllerTests.cs]: You don't have to specify the type you're `new`ing up if you've declared it on the lefthand side of the assignment.

So this line would be valid as:
```csharp
 private Division _division = new() { Id = 1, Name = "Test", Teams = new List<Team>()};
```
In fact, you can even simplify `new List<Team>()` as well:
```csharp
private Division _division = new() { Id = 1, Name = "Test", Teams = new() };
```

**Review by cszczepaniak — APPROVED** [review]
Small comments, but not gonna hold this one up.

File: CribblyBackend.UnitTests/ControllerTests/DivisionControllerTests.cs
Comment (adamszcz) [reply CribblyBackend.UnitTests/ControllerTests/DivisionControllerTests.cs]: Gotcha, that's neat. Fixed it!

**Pull request merged**
